ttnn/cpp/ttnn/tensor/tensor_layout.hpp

```diff
@@ -0,0 +1,102 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "types.hpp"
+#include "enum_types.hpp"
+
+#include <cstddef>
+#include <ostream>
+#include <variant>
+
+#include "shape.hpp"
+#include "alignment.hpp"
+#include "size.hpp"
+#include "page_config.hpp"
+
+namespace tt::tt_metal {
+
+using Strides = std::vector<size_t>;
```

Comment:
Shouldn't we use VectorBase here too, so we can allow for negative indexing?

Reply:
There was no need for it and I’d like to not use it without much clear need.